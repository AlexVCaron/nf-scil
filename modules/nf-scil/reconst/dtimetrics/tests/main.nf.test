nextflow_process {

    name "Test Process RECONST_DTIMETRICS"
    script "../main.nf"
    process "RECONST_DTIMETRICS"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "reconst"
    tag "reconst/dtimetrics"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-scil/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "heavy.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("reconst - dtimetrics") {

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/dwi/dwi.nii.gz", checkIfExists: true),
                        file("\${test_data_directory}/dwi/dwi.bval", checkIfExists: true),
                        file("\${test_data_directory}/dwi/dwi.bvec", checkIfExists: true),
                        []
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    niftiMD5SUM(process.out.ad.get(0).get(1), 8),
                    vectorMD5SUM(
                        process.out.evecs.get(0).get(1), process.out.tensor.get(0).get(1), 6),
                    vectorMD5SUM(
                        process.out.evecs_v1.get(0).get(1), process.out.tensor.get(0).get(1), 6),
                    vectorMD5SUM(
                        process.out.evecs_v2.get(0).get(1), process.out.tensor.get(0).get(1), 6),
                    vectorMD5SUM(
                        process.out.evecs_v3.get(0).get(1), process.out.tensor.get(0).get(1), 6),
                    niftiMD5SUM(process.out.evals.get(0).get(1), 8),
                    niftiMD5SUM(process.out.evals_e1.get(0).get(1), 8),
                    niftiMD5SUM(process.out.evals_e2.get(0).get(1), 8),
                    niftiMD5SUM(process.out.evals_e3.get(0).get(1), 8),
                    niftiMD5SUM(process.out.fa.get(0).get(1), 6),
                    niftiMD5SUM(process.out.ga.get(0).get(1), 6),
                    niftiMD5SUM(process.out.rgb.get(0).get(1), 8),
                    niftiMD5SUM(process.out.md.get(0).get(1), 8),
                    niftiMD5SUM(process.out.norm.get(0).get(1), 8),
                    niftiMD5SUM(process.out.rd.get(0).get(1), 8),
                    tensorMD5SUM(process.out.tensor.get(0).get(1), 7),
                    niftiMD5SUM(process.out.nonphysical.get(0).get(1), 8),
                    niftiMD5SUM(process.out.pulsation_std_dwi.get(0).get(1), 8),
                    niftiMD5SUM(process.out.residual.get(0).get(1), 2),
                    process.out.residual_iqr_residuals,
                    process.out.residual_mean_residuals,
                    process.out.residual_q1_residuals,
                    process.out.residual_q3_residuals,
                    process.out.residual_residuals_stats,
                    process.out.residual_std_residuals,
                    process.out.versions
                ).match() }
            )
        }

    }

    test("reconst - dtimetrics_with_b0mask") {

        config "./nextflow_light.config"

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/dwi/dwi.nii.gz", checkIfExists: true),
                        file("\${test_data_directory}/dwi/dwi.bval", checkIfExists: true),
                        file("\${test_data_directory}/dwi/dwi.bvec", checkIfExists: true),
                        file("\${test_data_directory}/dwi/mask.nii.gz", checkIfExists: true)
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    niftiMD5SUM(process.out.fa.get(0).get(1), 6),
                    process.out.versions
                ).match() }
            )
        }
    }
}
