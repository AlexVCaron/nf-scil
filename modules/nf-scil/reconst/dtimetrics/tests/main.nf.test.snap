{
    "reconst - dtimetrics_with_b0mask": {
        "content": [
            "test__fa.nii.gz:md5,4da8c2e248c0d93a018d0add9bbc6438",
            [
                "versions.yml:md5,f057a83131297624fb6d9db93df8d749"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-26T14:44:59.223632"
    },
    "reconst - dtimetrics": {
        "content": [
            "test__ad.nii.gz:md5,9aef42f05ef76cda519c1bf1a4d09e5a",
            "test__evecs.nii.gz:md5,5382bf25d965b3d08b1779b3c26542f3",
            "test__evecs_v1.nii.gz:md5,b3aafa3d64ab5189adbb1dd449aea59b",
            "test__evecs_v2.nii.gz:md5,de7388d34e37e843e39d7b39837fa4aa",
            "test__evecs_v3.nii.gz:md5,516d14d6647a03f4a2bd7f51cae20fce",
            "test__evals.nii.gz:md5,b5b6e6a3c13b72a89f4626169dc791a8",
            "test__evals_e1.nii.gz:md5,9aef42f05ef76cda519c1bf1a4d09e5a",
            "test__evals_e2.nii.gz:md5,d3b5247e8d017e737a3f32ccf6944a32",
            "test__evals_e3.nii.gz:md5,e749580da032647acf62dfb9405bcbda",
            "test__fa.nii.gz:md5,f17769290198fa8b258b228c591a799a",
            "test__ga.nii.gz:md5,8ce650bf62a5ff0979d29020ae4e3fa0",
            "test__rgb.nii.gz:md5,00f60c043c956b0e95393071a493df37",
            "test__md.nii.gz:md5,12baefb237c2b972cf0c6980b0c10681",
            "test__mode.nii.gz:md5,f6722dd9ae95c74358fd17dfa214f354",
            "test__norm.nii.gz:md5,69399c44ee62d1c52e271f39a1bac4be",
            "test__rd.nii.gz:md5,02afa7a113d7946f3a36a0365c1d5f26",
            "test__tensor.nii.gz:md5,91a3c8d626ffa8c99495a058b84eead0",
            "test__nonphysical.nii.gz:md5,63905c6dc26e33011608bba5f2c0e2f9",
            "test__pulsation_std_dwi.nii.gz:md5,fedf3ac5e312d43791269a9e4a18f48c",
            "test__residual.nii.gz:md5,eccbda5e154711cba41b3afc1365326d",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test__residual_iqr_residuals.npy:md5,4a71d46303bb3ac67088fe5cdaa5694a"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test__residual_mean_residuals.npy:md5,2247858eaf51c6e973649418f8721477"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test__residual_q1_residuals.npy:md5,4a71d46303bb3ac67088fe5cdaa5694a"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test__residual_q3_residuals.npy:md5,4a71d46303bb3ac67088fe5cdaa5694a"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test__residual_residuals_stats.png:md5,18ce14e9e554c8f5fb1cf0c87af36f9f"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test__residual_std_residuals.npy:md5,5e526c12b8a13b611bd1b2e00d4e1448"
                ]
            ],
            [
                "versions.yml:md5,f057a83131297624fb6d9db93df8d749"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0-rc1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-26T14:58:17.511641"
    }
}